apps/build-onchain-apps/contracts/src/SignatureMintERC721.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+import {NFTUtilities} from "./NFTUtilities.sol";
+
+// Errors
+error InsufficientFunds();
+
+/**
+ * @title SignatureMintERC721
+ * @dev contract that allows a user to mint a ERC721 for free with a cryptographically signed message.
+ *      This is useful for mints where you want to allow users to mint for free based on a signature from
+ *      your backend API.  This is an alternative approach to a merkel tree which is fully on-chain.
+ */
+contract SignatureMintERC721 is ERC721 {
+    using ECDSA for bytes32;
+    using MessageHashUtils for bytes32;
+
+    // Contract Params
+    uint256 public paidMintCost = 0.0001 ether;
+    address public freeMintSigner; // Wallet Address for the authorized wallet to distribute free mints.
```

Comment:
could be immutable.

Reply:
freeMintSigner is passed via constructor.. do you think we should hardcode in the contract itself?

Reply:
immutable args can be set in the constructor!

Reply:
hrm.. I can't see to get that to work:

![image](https://github.com/coinbase/build-onchain-apps/assets/7040142/aa1050ed-39d3-446d-9c2b-b3dd91b25002)

Reply:
fixed with immutable